### PR TITLE
Reroot treedata

### PR DESCRIPTION
**Pull request by brj1**
Hello Yu,

I've implemented the `reroot` function for the `treedata` class. Below is an example.

```r
library(ape)
library(phytools)
library(ggtree)
file <- system.file("extdata/BEAST", "beast_mcc.tree", package="treeio")
beast <- read.beast(file)
beast2 <- reroot(beast, 28)
multiplot(
    ggtree(beast) + geom_tiplab(aes(label=label)) + geom_tippoint(aes(subset=posterior==1)),
    ggtree(beast2) + geom_tiplab(aes(label=label)) + geom_tippoint(aes(subset=posterior==1)),
    ncol=2
)
```

I have noticed one problem with my implementation however. When the `reroot` function is first called it loads the ape and phytools packages which then override the `reroot` function causing further invocations to call `ape::reroot`. I am not sure how to fix this without loading ape and phytools first or using `ggtree::reroot`.

### CODE REVIEW

File: R/method-reroot.R [hunk 1]
```diff
@@ -22,3 +22,22 @@ setMethod("reroot", signature(object="phylo"),
...
+##' @exportMethod reroot
+setMethod("reroot", signature(object="treedata"),
+        function(object, node, ...) {
+        	# reroot tree
+            tree <- object@phylo
+            tree <- reroot(tree, node, ...)
+            object@phylo <- tree
+            
+            # update node numbers in data
+            n.tips <- length(tree$tip.label) # Is there a better way in ggtree/treeio to get the number of tips?
```
Comment (GuangchuangYu) [inline R/method-reroot.R]: Ntip(tree)

File: R/method-reroot.R [hunk 2]
```diff
@@ -22,3 +22,22 @@ setMethod("reroot", signature(object="phylo"),
...
+            node_map <- attr(tree, "node_map")
+            data <- object@data
+            data$node[match(node_map$from, as.integer(data$node))] <- node_map$to
```
Comment (GuangchuangYu) [inline R/method-reroot.R]: If object@extraInfo has information, also need to map.

Comment (brj1) [reply R/method-reroot.R]: Thanks for the input Yu. I will implement all of these in some time.

Comment (brj1) [reply R/method-reroot.R]: Hi Yu, Can you point me to an example of treedata with `@extraInfo`? I'm not sure what it looks like.

Comment (GuangchuangYu) [reply R/method-reroot.R]: similar to `@data`, and both can be empty. <https://github.com/GuangchuangYu/treeio/blob/5900d808d65f1cdd6ffdc277d361618c8c905163/R/method-drop-tip.R#L72-L84>

Comment (brj1) [reply R/method-reroot.R]: Thanks. I've implemented updating `object@extraInfo` and changed the update of `object@data` to be similar.

**Comment by GuangchuangYu** [comment]
thanks!

BTW: I may move this method to `treeio` package. I think it is more suitable to be paced on `treeio`.

**Pull request merged**